### PR TITLE
Add clearMessages method to chat reference

### PR DESCRIPTION
**Pull request by JamyDev**

### CODE REVIEW

**Comment by connor4312** [comment]
lgtm after the presumably-incoming PR for the chat server itself 😛

File: src/reference/chat/data.json [hunk 1]
```diff
@@ -174,6 +174,23 @@
         }
       }
     },
+    "clearMessages": {
+      "description": "Clears all chat messages in the channel. (If you have the `clear_messages` permission)",
```
Comment (connor4312) [inline src/reference/chat/data.json]: Might be useful to have an annotation for permissions in the chat server methods

**Comment by JamyDev** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;JPHjsy4e&#x2F;118-add-clear-method-to-chat-server"> Add clear method to chat server</a>

File: src/reference/chat/data.json (hunk 1 above)
Comment (JamyDev) [inline src/reference/chat/data.json]: Something @ProbablePrime could perhaps look into? :)

**Comment by ProbablePrime** [comment]
Let's merge this and factor permissions in later.

**Pull request merged**